### PR TITLE
cpu/esp_common: use generic WIFI_SSID/WIFI_PASS defines

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Writing generic applications that work with both `esp_wifi` and `atwinc15x0` is hampered by the different names for the SSID/Password defines.

`esp_wifi` uses `ESP_WIFI_SSID` while `atwinc15x0` uses the more generic `WIFI_SSID`.

Also use the generic name for `esp_wifi`. 

### Testing procedure

Connecting to a WiFi AP should still work.

### CODE REVIEW

**Review by maribu — APPROVED** [review]

File: cpu/esp32/doc.txt [hunk 1]
```diff
@@ -1612,7 +1612,7 @@ following configuration parameters have to be defined:
 
 Parameter           | Default   | Description
 :-------------------|:----------|:------------
-#ESP_WIFI_SSID      | "RIOT_AP" | SSID of the AP to be used.
+#WIFI_SSID      | "RIOT_AP" | SSID of the AP to be used.
 ESP_WIFI_EAP_ID     | none      | Optional anonymous identity used in phase 1 (outer) EAP authentication.[1]
 ESP_WIFI_EAP_USER   | none      | User name used in phase 2 (inner) EAP authentication.
 ESP_WIFI_EAP_PASS   | none      | Password used in phase 2 (inner) EAP authentication.
```
Comment (gschorcht) [inline cpu/esp32/doc.txt:1618]: The question is whether all other WIFI settings that are not specific to ESP should not also be changed.
```suggestion
#WIFI_SSID      | "RIOT_AP" | SSID of the AP to be used.
WIFI_EAP_ID     | none      | Optional anonymous identity used in phase 1 (outer) EAP authentication.[1]
WIFI_EAP_USER   | none      | User name used in phase 2 (inner) EAP authentication.
WIFI_EAP_PASS   | none      | Password used in phase 2 (inner) EAP authentication.
```

File: cpu/esp32/doc.txt [hunk 2]
```diff
@@ -1630,7 +1630,7 @@ line, for example:
 
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 USEMODULE=esp_wifi_enterprise \
-CFLAGS='-DESP_WIFI_SSID=\"MySSID\" -DESP_WIFI_EAP_ID=\"anonymous\" -DESP_WIFI_EAP_USER=\"MyUserName\" -DESP_WIFI_EAP_PASS=\"MyPassphrase\"' \
+CFLAGS='-DWIFI_SSID=\"MySSID\" -DESP_WIFI_EAP_ID=\"anonymous\" -DESP_WIFI_EAP_USER=\"MyUserName\" -DESP_WIFI_EAP_PASS=\"MyPassphrase\"' \
```
Comment (gschorcht) [inline cpu/esp32/doc.txt:1633]: The same here.
```suggestion
CFLAGS='-DWIFI_SSID=\"MySSID\" -DWIFI_EAP_ID=\"anonymous\" -DWIFI_EAP_USER=\"MyUserName\" -DWIFI_EAP_PASS=\"MyPassphrase\"' \
```

File: cpu/esp_common/esp-wifi/doc.txt [hunk 3]
```diff
@@ -87,7 +87,7 @@ following configuration parameters have to be defined:
 
 Parameter          | Default   | Description
 :------------------|:----------|:------------
-ESP_WIFI_SSID      | "RIOT_AP" | SSID of the AP to be used.
+WIFI_SSID      | "RIOT_AP" | SSID of the AP to be used.
 ESP_WIFI_EAP_ID    | none      | Optional anonymous identity used in phase 1 (outer) EAP authentication. If it is not defined, the user name defined for phase 2 (inner) EAP authentication is used as idendity in phase 1.
 ESP_WIFI_EAP_USER  | none      | User name used in phase 2 (inner) EAP authentication.
 ESP_WIFI_EAP_PASS  | none      | Password used in phase 2 (inner) EAP authentication.
```
Comment (gschorcht) [inline cpu/esp_common/esp-wifi/doc.txt:93]: The same here:
```suggestion
WIFI_SSID      | "RIOT_AP" | SSID of the AP to be used.
WIFI_EAP_ID    | none      | Optional anonymous identity used in phase 1 (outer) EAP authentication. If it is not defined, the user name defined for phase 2 (inner) EAP authentication is used as idendity in phase 1.
WIFI_EAP_USER  | none      | User name used in phase 2 (inner) EAP authentication.
WIFI_EAP_PASS  | none      | Password used in phase 2 (inner) EAP authentication.
```

File: cpu/esp_common/esp-wifi/doc.txt [hunk 4]
```diff
@@ -101,7 +101,7 @@ line, for example:
 
 ```
 USEMODULE=esp_wifi_enterprise \
-CFLAGS='-DESP_WIFI_SSID=\"MySSID\" -DESP_WIFI_EAP_ID=\"anonymous\" -DESP_WIFI_EAP_USER=\"MyUserName\" -DESP_WIFI_EAP_PASS=\"MyPassphrase\"' \
+CFLAGS='-DWIFI_SSID=\"MySSID\" -DESP_WIFI_EAP_ID=\"anonymous\" -DESP_WIFI_EAP_USER=\"MyUserName\" -DESP_WIFI_EAP_PASS=\"MyPassphrase\"' \
```
Comment (gschorcht) [inline cpu/esp_common/esp-wifi/doc.txt:104]:
```suggestion
CFLAGS='-DWIFI_SSID=\"MySSID\" -DWIFI_EAP_ID=\"anonymous\" -DWIFI_EAP_USER=\"MyUserName\" -DWIFI_EAP_PASS=\"MyPassphrase\"' \
```

**Comment by maribu** [comment]
bors merge

**Review by gschorcht — CHANGES_REQUESTED** [review]

**Comment by maribu** [comment]
bors cancel

**Comment by benpicco** [comment]
> I would like also to generalize the WPA2 Enterprise configurations.

sure, added

**Comment by gschorcht** [comment]
Since we have been using the `ESP_WIFI_*` definitions in RIOT for the last 4 years and have not marked them as deprecated yet, we should add some default settings for changed configurations like
```
#if !defined(WIFI_SSID) and defined(ESP_WIFI_SSID)
#define WIFI_SSID ESP_WIFI_SSID
#endif
```
At least we should add something like
```
#if defined(ESP_WIFI_SSID)
#warning "ESP_WIFI_SSID is obsolete, please use WIFI_SSID instead"
#endif
```

File: cpu/esp_common/esp-wifi/esp_wifi_params.h [hunk 5]
```diff
@@ -44,15 +44,23 @@
 /**
  * @brief   SSID of the AP to be used.
  */
-#ifndef ESP_WIFI_SSID
-#define ESP_WIFI_SSID       "RIOT_AP"
+#ifndef WIFI_SSID
+#ifdef ESP_WIFI_SSID
+#define WIFI_SSID           ESP_WIFI_SSID
```
Comment (benpicco) [inline cpu/esp_common/esp-wifi/esp_wifi_params.h:49]: Already done here

File: cpu/esp_common/esp-wifi/esp_wifi_params.h [hunk 6]
```diff
@@ -44,15 +44,23 @@
...
 /**
  * @brief   Passphrase used for the AP as clear text (max. 64 chars).
  */
 #ifdef DOXYGEN
-#define ESP_WIFI_PASS       "ThisistheRIOTporttoESP"
+#define WIFI_PASS           "ThisistheRIOTporttoESP"
+#endif
+
+#if !defined(WIFI_PASS) && defined(ESP_WIFI_PASS)
+#define WIFI_PASS           ESP_WIFI_PASS
```
Comment (benpicco) [inline cpu/esp_common/esp-wifi/esp_wifi_params.h:63]: and here

Comment (gschorcht) [reply cpu/esp_common/esp-wifi/esp_wifi_params.h:63]: Ah ok :+1:

File: cpu/esp_common/esp-wifi/esp_wifi_netdev.c [hunk 7]
```diff
@@ -947,20 +947,20 @@ void esp_wifi_setup (esp_wifi_netdev_t* dev)
 
 #if defined(MODULE_ESP_WIFI_ENTERPRISE) && !defined(MODULE_ESP_WIFI_AP)
 
-#ifdef ESP_WIFI_EAP_ID
-    esp_wifi_sta_wpa2_ent_set_identity((const unsigned char *)ESP_WIFI_EAP_ID,
-                                       strlen(ESP_WIFI_EAP_ID));
-#endif /* ESP_WIFI_EAP_ID */
-#if defined(ESP_WIFI_EAP_USER) && defined(ESP_WIFI_EAP_PASS)
+#ifdef WIFI_EAP_ID
```
Comment (gschorcht) [inline cpu/esp_common/esp-wifi/esp_wifi_netdev.c:950]: We should have some default definitions here for compatibility reasons:
```suggestion

#if !defined(WIFI_EAP_ID) && defined(ESP_WIFI_EAP_ID)
#define WIFI_EAP_ID     ESP_WIFI_EAP_ID
#endif

#if !defined(WIFI_EAP_USER) && defined(ESP_WIFI_EAP_USER)
#define WIFI_EAP_USER   ESP_WIFI_EAP_USER
#endif

#if !defined(WIFI_EAP_PASS) && defined(ESP_WIFI_EAP_PASS)
#define WIFI_EAP_PASS   ESP_WIFI_EAP_PASS
#endif

#ifdef WIFI_EAP_ID
```

File: cpu/esp_common/esp-wifi/esp_wifi_netdev.c [hunk 8]
```diff
@@ -947,6 +947,19 @@ void esp_wifi_setup (esp_wifi_netdev_t* dev)
 
 #if defined(MODULE_ESP_WIFI_ENTERPRISE) && !defined(MODULE_ESP_WIFI_AP)
 
+
```
Comment (gschorcht) [inline cpu/esp_common/esp-wifi/esp_wifi_netdev.c:950]: Sorry, there was an empty line too much in my suggestion.
```suggestion
```

**Comment by gschorcht** [comment]
Please squash.

**Review by gschorcht — APPROVED** [review]

**Comment by gschorcht** [comment]
bors merge

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
bors cancel
bors merge

**Comment by benpicco** [comment]
bors merge

**Pull request merged**